### PR TITLE
Salesforce connect

### PR DESCRIPTION
**Pull request by harshkothari410**

### CODE REVIEW

**Comment by harshkothari410** [comment]
- Added Salesforce connection capacity using nforce package 
- Added test for Argus connection

I am not able to add test for salesforce connectivity because we are using nforce package and it try to connect actual salesforce end points which I cant specify so I am not able to test using nock. @iamigo

File: src/remoteCollection/collect.js [hunk 1]
```diff
@@ -73,15 +74,18 @@ function sendRemoteRequest(generator) {
...
-          return prepareRemoteRequest(generator)
-            .then((resp) => {
-              if (resp) {
-                debug('sendRemoteRequest returned OK');
-                generator.res = resp.res;
-              }
-
-              return resolve(generator);
-            });
+          prepareRemoteRequest(generator)
```
Comment (iamigo) [inline src/remoteCollection/collect.js]: are you sure *removing* the return from this call is correct??

File: test/remoteCollection/collect.js [hunk 2]
```diff
@@ -162,6 +162,87 @@ describe('test/remoteCollection/collect.js >', () => {
       .catch(done);
     });
 
+    it('collecting data from argus', (done) => {
```
Comment (iamigo) [inline test/remoteCollection/collect.js]: no test for the "connecting to salesforce" functionality being added in this pr?

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: Check my above comment @iamigo

Comment (iamigo) [reply test/remoteCollection/collect.js]: maybe take a look at how the nforce module runs its tests to see if we can leverage something from there? https://github.com/kevinohara80/nforce/tree/master/test

Comment (iamigo) [reply test/remoteCollection/collect.js]: oh, that's too bad :(

ok, so we'll have to run manual tests for this until we can figure something out

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: Yes I have opened few issue on wreck and simple-oauth2. If they fix that then we will remove nforce.

File: src/remoteCollection/collect.js [hunk 3]
```diff
@@ -41,7 +42,7 @@ function sendRemoteRequest(generator) {
 
     // If token is present, add to request header.
     if (generator.token) {
-      const accessToken = generator.token.accessToken;
+      const accessToken = generator.token.accessToken || generator.token.access_token;
```
Comment (cassiodias) [inline src/remoteCollection/collect.js]: is this a valid expectation? didn't find that in the docs or tests

Comment (harshkothari410) [reply src/remoteCollection/collect.js]: Its universal standard either accessToken or access_token will be returned from remote source.

File: test/remoteCollection/collect.js (hunk 2 above)
Comment (jgraff2) [reply test/remoteCollection/collect.js]: Can you look at the source code to find the hardcoded url, then mock that?

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: They are trying to hit that URL so I cant moke behalf of them.

Comment (jgraff2) [reply test/remoteCollection/collect.js]: I don't understand. 
Do we know what the url is?
Do we know what the response from that url will be?
If so, we should be able to mock it, right?

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: I cant mock behalf of them. I am not posting request. Request is done by them. Give me example how to do this. I have no idea how to do this.

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: screen shot 2018-10-25 at 2 46 28 pm

I tried this. This thing is not working. So if you have some other idea let me know.

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: screen shot 2018-10-25 at 2 49 00 pm
I tried this but its not working . SO if you have any other ideas let me know.

Comment (harshkothari410) [reply test/remoteCollection/collect.js]: Let me know this looks good or not. I tried to add test but not sure this is you want @jgraff2 .

Comment (jgraff2) [reply test/remoteCollection/collect.js]: @harshkothari410 looks good, that's what I was thinking.

**Pull request merged**